rust-escrow/stable-escrow/src/escrow.rs

```diff
@@ -0,0 +1,229 @@
+     * Transfer self NEP141 in exchange of the stable NEP141 amount as a receipt
+     */
+    #[private]
+    pub fn deposit(&mut self, sender_id: AccountId, amount: Balance) {
+        if !self.is_deposit_allowed() {
+            env::panic_str("ERR_DEPOSIT_NOT_ALLOWED");
+        }
+
+        if amount > self.get_unpaid_amount() {
+            env::panic_str("ERR_DEPOSIT_NOT_ALLOWED");
+        }
+
+        self.ft.internal_deposit(&sender_id, amount);
+        self.deposits.insert(&sender_id);
+        self.metadata.unpaid_amount = self
+            .metadata
+            .unpaid_amount
+            .checked_sub(amount)
+            .unwrap_or_else(|| env::panic_str("ERR_UNPAID_AMOUNT_OVERFLOW"));
+
+        // @TODO log
+    }
+
+    /**
+     * Transfer funds from contract NEP141 balance to sender_id
+     */
+    #[payable]
+    pub fn withdraw(&mut self) -> Promise {
+        if !self.is_withdrawal_allowed() {
+            env::panic_str("ERR_WITHDRAWAL_NOT_ALLOWED");
+        }
+
+        let payee = env::signer_account_id();
+        let balance = self.ft.internal_unwrap_balance_of(&payee);
+
+        // Transfer from collateral token to payee
+        let promise = Promise::new(self.metadata.nep_141.clone()).function_call(
+            "ft_transfer".to_string(),
+            json!({
+                "amount": balance.to_string(),
+                "receiver_id": payee.to_string(),
+            })
+            .to_string()
+            .into_bytes(),
+            1, // 1 yoctoNEAR
+            GAS_FT_TRANSFER,
+        );
+
+        let callback = Promise::new(env::current_account_id()).function_call(
+            "on_withdraw_callback".to_string(),
+            json!({
+                "payee": payee.to_string(),
+                "balance": balance.to_string(),
+            })
+            .to_string()
+            .into_bytes(),
+            0,
+            GAS_FT_WITHDRAW_CALLBACK,
+        );
+
+        promise.then(callback)
+    }
+
+    /**
+     * Only if total funds are reached, allow to call this function
+     * Transfer total NEP141 funds to a new DAO
+     * Make the depositors members of the DAO
+     */
+    #[payable]
+    pub fn delegate_funds(&mut self) -> Promise {
+        if self.is_deposit_allowed() || self.is_withdrawal_allowed() {
+            env::panic_str("ERR_DELEGATE_NOT_ALLOWED");
+        }
+
+        // env::panic_str("ERR_TOTAL_FUNDS_OVERFLOW");
+
+        // @TODO charge a fee here (1.5% initially?) when a property is sold by our contract
+
+        let dao_name = format!("{}", self.metadata.id);
+        let args =
+            self.get_dao_config(dao_name.clone(), vec![self.metadata.maintainer.to_string()]);
+
+        let promise = Promise::new(self.get_dao_factory_account_id()).function_call(
+            "create".to_string(),
+            json!({ "name": dao_name.clone(), "args": Base64VecU8(args) })
+                .to_string()
+                .into_bytes(),
+            BALANCE_FOR_CREATE_DAO,
```

Comment:
@susguzman I think this balance will need to be passed from `env::attached_deposit` so that we control this amount from the UI.

Reply:
+1